src/sources/kubernetes_logs/mod.rs

```diff
@@ -622,6 +663,9 @@ impl Source {
             NamespaceMetadataAnnotator::new(ns_state, namespace_fields_spec, log_namespace);
         let node_annotator = NodeMetadataAnnotator::new(node_state, node_field_spec, log_namespace);
 
+        let ignore_before =
+            ignore_older_secs.map(|secs| Utc::now() - chrono::Duration::seconds(secs as i64));
+
```

Comment:
I'm wondering if we want to move this into the `file_server` itself, as both the `file` and `kubernetes_logs` sources are doing this same calculation.

Reply:
Good point! In which way should we implement it? I suggest implementing it as a free function in `file_server` corresponding file and calling this function in `file` and `kubernetes` sources. What do you think?

Reply:
👍 I think that makes sense to me!

Reply:
Done